File: Week02/Day2/diagonal_matrix.py
# - Create (dynamically) a two dimensional list
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional list to the output

l = [1, 0, 0, 0]

def create_grid(grid_height, grid_width):
    grid = []
    for i in range(grid_height):
        row_list = []
        for j in range(grid_width):
            if j == i:
                row_list.append("1")
            else:    
                row_list.append ("0")
        grid.append (row_list)
    return grid

def print_grid(grid):
    for row in grid:
        print(row)

print_grid(create_grid(4,4))

File: Week07/Day3/diagonal_matrix.js
'use strict';
// - Create (dynamically*) a two dimensional list
//   with the following matrix**. Use a loop!
//
//   0 0 0 1
//   0 0 1 0
//   0 1 0 0
//   1 0 0 0
//
// - Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array

function leftMatrix (size) {
    let row = []
    for (let i = 0; i < size; i++) {
        for (let j = 0; j < size; j++) {
            if (j === i) {
                row.push (1)
            } else {
                row.push(0)
            }
        } console.log(row)
        row = []
        }
}

leftMatrix(4)

console.log('')

// for loops can be the same as above

function rightMatrix (size) {
    let row = []
    for (let i = size; i > 0; i--) {
        for (let j = size; j > 0; j--) {
            if (j === i) {
                row.splice (0, 0, 1)
            } else {
                row.splice(0, 0, 0)
            }
        } console.log(row)
        row = []
        }
}

rightMatrix(4)